rabbitmq_client/publisher.py

```diff
@@ -34,27 +36,51 @@ def register_serializer(self, func: Callable):
             raise DuplicateSerializerError(f"Serializer already assigned with name '{self.serializer.__name__}'")
         self.serializer = func
 
-    def publish(self, queue_config: PublishQueueConfig, payload: dict, headers: Optional[dict] = None,
-                priority: Optional[int] = 0):
+    def publish(self, queue_config: PublishQueueConfig,
+                payload: Optional[dict] = None, payload_list: List[dict] = None,
+                headers: Optional[dict] = None, max_retries: int = 3, priority: Optional[int] = 0):
         if headers is None:
             headers = {}
         assert isinstance(queue_config, PublishQueueConfig), \
             f"Expected instance of PublishQueueConfig, passed {type(queue_config)}"
-        assert isinstance(payload, dict), \
-            f"Expected instance of dict, passed {type(payload)}"
-        assert isinstance(headers, dict), \
-            f"Expected instance of dict, passed {type(headers)}"
+        assert self.serializer is not None, \
+            "Define custom serializer with @publish.register_serializer decorator"
+        assert isinstance(headers, dict), f"Expected instance of dict, passed {type(headers)}"
         assert isinstance(priority, int), \
             f"Expected instance of int, passed {type(priority)}"
 
-        serializer: Callable = self.serializer or default_serializer
-        stringified_payload = json.dumps(payload, default=serializer).encode('utf-8')
+        # Check that either payload or payload_list is provided, but not both None
+        if payload is None and (payload_list is None or len(payload_list) == 0):
+            raise ValueError("Either 'payload' or 'payload_list' must be provided.")
+
+        if payload_list is None or len(payload_list) == 0:
+            payload_list = [payload]
+
+        for payload_item in payload_list:
+            assert isinstance(payload_item, dict), \
+                f"Expected instance of dict, passed {type(payload_item)}"
```

Comment:
can this be done at the time of publish?
so that we dont' have to run the loop twice

Reply:
Ideally we should do validations separately from business logic.